rustls/src/lib.rs

```diff
@@ -336,6 +336,9 @@ mod x509;
 mod check;
 mod bs_debug;
 mod builder;
+#[allow(non_camel_case_types)]
```

Comment:
If we need these, I think they should go in an inner attribute inside the module. Why do we need them?

Reply:
I started down this path, and realized that the comments would be pointless repetition of the enum contents. For instance, commenting on `SSLv2` with "SSL Version 2" doesn't really help anybody. Similarly, commenting `TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256` with some prose explanation of its components would be pointless. The RFCs don't attempt to do that when registering the parameters.

Reply:
Fair enough, but maybe the enums themselves should get a short docstring pointing to the RFC/IANA registry or whatever?

Reply:
Would still be good to move these to inner attributes IMO.

Reply:
I tried putting `#[allow(non_camel_case_types)]` at the top of rustls/src/enums.rs, but it did not appear to take effect for the declarations within the file.

I tried putting that attribute directly above the `enum_builder!` calls, but it similarly did not appear to take effect.

I tried putting that attribute inside the definition of `enum_builder!`, under the logic that most of the enums are non-camel case. But some (like HandshakeType) _are_ camel case, so I got "warning: unused attribute `allow`".

I'm sure there's a way to do this and I'll keep poking at it, but if you have an idea of how it should work, let me know.

Reply:
It needs to become an **inner** attribute, I think? `#![allow(non_camel_case_types)]`.

Reply:
Ah, thanks for that. I hadn't grokked the difference between inner and outer attributes.